src/sgame/sg_struct.h

```diff
@@ -74,10 +74,10 @@ struct gentityConditions_t
 	team_t   team;
 	int      stage;
 
-	class_t     classes[ PCL_NUM_CLASSES ];
-	weapon_t    weapons[ WP_NUM_WEAPONS ];
-	upgrade_t   upgrades[ UP_NUM_UPGRADES ];
-	buildable_t buildables[ BA_NUM_BUILDABLES ];
+	BoundedVector<class_t,     PCL_NUM_CLASSES>   classes;
```

Comment:
To expand on why introducing a class-memaccess warning is bad: the warning doesn't tell you which field of the struct generated the error, so you have to manually inspect all the fields to see which one it is and check whether it is a true positive.

Reply:
Fixed